packages/wrangler/src/dev/remote.tsx

```diff
@@ -323,10 +323,13 @@ export function useWorker(
 					port: workerPreviewToken.inspectorUrl.port.toString(),
 					pathname: workerPreviewToken.inspectorUrl.pathname,
 				},
-				userWorkerInnerUrlOverrides: {}, // there is no analagous prop for this option because we did not permit overriding request.url in remote mode
+				userWorkerInnerUrlOverrides: {
+					hostname: props.host,
+					port: props.port.toString(),
+				},
 				headers: {
 					"cf-workers-preview-token": workerPreviewToken.value,
-					Cookie: accessToken && `CF_Authorization=${accessToken}`,
+					...(accessToken ? { Cookie: `CF_Authorization=${accessToken}` } : {}),
```

Comment:
I noted that we had to make the `headers` record allow `undefined` values just to accommodate this `Cookie` assignment. But in reality we don't actually want to assign `undefined` to Cookie, since the code later one would result in the Cookie looking like: `previous-cookies;undefined`.

So I fixed this up here, removed the `undefined` from the type and was able to avoid a `!` operator later on.

Reply:
This is fair enough. And is probably the "correct" solution. But this type will eventually be exposed to users to write themselves, and I believe allowing `undefined` will be more ergonomic and subtly improve DX

Reply:
I'm not convinced that allowing `undefined` here adds to user DX. If anything it creates a mental overhead because the user has to think about what setting to `undefined` actually means (when it actually means ignore this header and not remove the original header).